### ISSUE TITLE
Use the Persons Feature to impersonate users as the STFC

### ISSUE THREAD

**Issue by TCMeldrum** [UserOfficeProject/user-office-project-issue-tracker#1100]
We would like to be able to log in a user and see what a user sees. There is the person feature that exists, we can turn on this feature on and see if it works.

Areas that might make this difficult: how we interact with the uows when we are impersonating a user
The person feature page (if i remember correctly) might use a lot more data than we have stores in the db so im not sure how that page will work.

Feature requested by George through teams and other convocations. 

Before we can use this we need to make sure the following are done:

- [x] #1406
- [x] Workout if there is anything a impersonating user cannot do they - from the way it is implemented I think that they can do anything the impersonated user can do
- [x] #1405
- [x] #1413

**Comment by joshhdawes** [comment]
@georgeflecknell while this will be really helpful for you, and will probably save a lot of time, as far as I know there's no deadline on this imposed by stakeholders. Is this the case?

**Comment by joshhdawes** [comment]
@georgeflecknell following up - could you please let me know if there's a deadline for this, or if it's just a nice-to-have?

**Comment by georgeflecknell** [comment]
That's definitely a nice to have.

**Comment by joshhdawes** [comment]
Comments from cross-team on 4/12/24:

- Currently behind a feature flag - can just be enabled
- Need to show that any changes made as a user are recorded as having been made in "impersonate" mode.
- Should change nav bar colour to indicate "impersonate" mode

Should let Emma know when we turn this on (only turn it on when we need to)

**Comment by joshhdawes** [comment]
Comments from meeting with @Cat-Lucifer, @srconway, @georgeflecknell, and @TCMeldrum on 23/5/25:

- Can be turned on, but logging needs to be sufficient to allow us to see whether a change was made by a user, or by a user officer impersonating a user. For now, this will probably be done by logging the start and end of an impersonation session.
- @Cat-Lucifer is going to help thin-out the list of all users with user officer roles.
- Actions need to be identified that impersonated users can't perform.
- We'll need some documentation for this, which should include any side-effects and the actions that can be performed when impersonating a user.

Note: once finalised, this will be a major help when dealing with problems reported by users.

**Comment by srconway** [comment]
Before we enable this we've checked who has User Officer rights and identified a few who should be removed:

2 x ex-User Office staff - I've emailed FIT to remove them - confirmed as done
3 x CLF FAP groups. We can't remove the people from the groups but we should unlink the groups from the role
2 x ISIS Directors - we need to include them in requirements around management overview rights  in the longer term
ISIS Developers group - this includes all developers including those still on probation etc. On Production we should limit this to ISIS Support Developers which only includes people who (should) have the experience to know what they're doing. (On Dev ISIS Developers would be the right group to use)

**Comment by joshhdawes** [comment]
@mehta-pooja123 tagging you in this since you're currently interested in improving how roles/permissions work in UOP. I think, for us to ensure that the correct people can use this feature, some work might have to be done on how roles work (to have one User Officer be allowed to use this, and another not).

**Comment by srconway** [comment]
I agree, it feels like this should be a permission in its own right - a kind of Super-user-officer. That doesn't really fit with the current roles where you only get the rights from one role at at time, whereas ideally you'd be allowed the combination of rights gained from all the roles you hold.

In the short term can we at least unlink the FAP groups and Developers group from the User Officer role on Prod before we enable impersonation (Developers at least will need to stay linked on Dev, is that feasible?)

**Comment by TCMeldrum** [comment]
The feature is now offically avalible on prod (it kind of always was but we have now told people they can start using it). Are we happy to close this issue and deal with the permissions issue seperatly?
